Сетка страниц и крупных блоков на флексах

PolyTS opened the pull request:
---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/35/user/2075923/tasks/10)

:boom: https://htmlacademy-htmlcss.github.io/2075923-device-35/5/

PolyTS committed:
Сетка страниц и крупных блоков на флексах

pavel-niukalo commented on index.html:
```diff
@@ -74,90 +75,96 @@
-    <main>
-      <h1 class="visually-hidden">Интернет-магазин техники Device</h1>
+    <main class="main-conteiner">
+      <h1 class="visually-hidden" hidden>Интернет-магазин техники Device</h1>
```
Атрибут `hidden` скроет элемент из дерева доступности, что нам не надо, поэтому тут можно обойтись только классом `visually-hidden`

pavel-niukalo commented on index.html:
```diff
@@ -74,90 +75,96 @@
+          </ul>
+        </div>
       </section>
+      <h2 class="visually-hidden" hidden>Меню товаров</h2>
```
Тут так же про `hidden`, и второй момент, раньше у нас этот заголовок находился в секции, а сейчас как бы "оторван". Лучше вернуть в начало секции как было

pavel-niukalo commented on styles/styles.css:
```diff
@@ -14,34 +14,126 @@
 body {
+  display: flex;
+  flex-direction: column;
+  width: 1440px;
```
`body` нам не нужно ограничивать, поскольку фоновые цвета, которые упираются в края макета, должны тянуться на всю ширину экрана

![image](https://user-images.githubusercontent.com/52506235/198886444-f4cb5652-d1d6-4498-b60d-607b875d2b88.png)

То есть, если сайт открыт на экране с шириной в 1980px, то `body` должен быть во всю ширину, а фон должны тянуться. Вот таких белых полос не должно быть

![image](https://user-images.githubusercontent.com/52506235/198886495-1ad8e1b1-030a-423c-b6f5-fdb3958a6d68.png)

pavel-niukalo commented on styles/styles.css:
```diff
@@ -14,34 +14,126 @@
   list-style-type: none;
 }
 
+/*выпадающее меню.изначално у меня это один список, но теперь думаю, стоит ли их поделить на три списка, чтобы могла каждому списку задать размер?или есть другой способ внутри списка перераспределять положение.*/
```
В целом, мы можем задать `flex-direction: column`, ограничить высотой и `flex-wrap` будет переносить нам элементы вправо. Но, тут у нас интересное расположение - лесенкой

![image](https://user-images.githubusercontent.com/52506235/198886788-305c5a45-8d94-4ae4-9fce-8bc886c538da.png)

И если сделать как я описал выше, то по середине у нас еще будет место и последний элемент перенесется в это место

![image](https://user-images.githubusercontent.com/52506235/198886831-b4d165d6-1b21-4256-b53b-41f810b48b72.png)

Поэтому тут да, имеет смысл сделать три списка

![image](https://user-images.githubusercontent.com/52506235/198886883-3d0cc3c7-1bd6-4d43-b16d-e5a93c7a357d.png)

pavel-niukalo commented on styles/styles.css:
```diff
@@ -340,15 +581,27 @@ body {
+}
+
 .inner-header-title {
+  width: 541;
```
Невалидное значение - не указаны единицы измерения

PolyTS committed:
Доп.правки "Сетка страниц и крупных блоков на флексах"

1)убрала атрибут hidden
2)убрала фикс.ширину у <body>.Поправила поведение фонов, которые упираются в края макета.
3)сделала три списка в выпадающем меню.
4)добавила пропущенное значение
